src/workloads/execution/clusteredCollection.yml

```diff
@@ -0,0 +1,77 @@
+SchemaVersion: 2018-07-01
+Owner: "@mongodb/server-execution"
+Description: >
+  Run basic insert and find workload on a collection clustered by {_id: 1} .
+  Clustered collections are planned to mainly serve operations over the cluster
+  key such as monotonically increasing inserts, range queries and range deletions.
+
+Actors:
+
+# Phase 1: create collection
+- Name: Create
+  Type: RunCommand
+  Threads: 1
+  Phases:
+  - Repeat: 1
+    Database: test
+    Operations:
+    - OperationName: RunCommand
+      OperationCommand:
+        create: Collection0
+        clusteredIndex: {key: {_id: 1}, unique: true}
+  - &Nop {Nop: true}
+  - *Nop
+  - *Nop
+
+# Phase 2: insert. TODO Make multithreaded once TIG-2938 is resolved.
+- Name: Insert
+  Type: Loader
+  Threads: 1
+  Phases:
+  - *Nop
+  - Repeat: 1
+    Database: test
+    Threads: 1
+    CollectionCount: 1
+    DocumentCount: &docCount 1000000
+    BatchSize: 1000
+    Document:
+      _id: {^Inc: {start: 0}}
```

Comment:
Open to suggestions on a representative _id type and size. Perhaps `Timestamp` for the likes of change collections? Also, do we plan to see performance-critical RecordId's much larger than 128 bytes (AFAIK the largest RecordId is for pre-image collections and is _only_ 136 bytes) in the future?

Reply:
You could use a combination of `^ObjectId` / `^FormatString` to generate whatever format you wanted.
For Example:

```
  - Name: FormatString ^Inc
    GivenTemplate:
      a: {^ObjectId: {^FormatString: {"format": "c%03daaaaaaaaaaaaaaaaaaaa", "withArgs": [{^Inc: {start: 0, step: 1, multiplier: 0}}]}}}
    ThenReturns:
      - { "a" : { "$oid" : "c000aaaaaaaaaaaaaaaaaaaa"} }
      - { "a" : { "$oid" : "c001aaaaaaaaaaaaaaaaaaaa"} }
      - { "a" : { "$oid" : "c002aaaaaaaaaaaaaaaaaaaa"} }
      - { "a" : { "$oid" : "c003aaaaaaaaaaaaaaaaaaaa"} }

```

There are some random examples [here](https://github.com/mongodb/genny/blob/809d76b4e83bea8537d997954c2a30cbe84e9479/src/workloads/issues/MongosLatency.yml#L29).

Reply:
Thanks Jim, that example is very useful. I've parametrised this file and used `^FormatString` to run the workload with different RecordId sizes.